src/main/java/com/avioconsulting/log4j/sqs/client/ConnectorClient.java

```diff
@@ -78,29 +77,41 @@ private void initSQSClient(ConnectorClientAttributes attributes) {
 		}
 	}
 
-
-	public String getQueueURL(String queueName) {
-		if(this.queueUrl == null) {
-			this.queueUrl = this.awsSQSAsyncClient.getQueueUrl(queueName).getQueueUrl();
+	public void sendMessages(MessageRequestWrapper messageList, String processorType, String queueName, String largeMessageQueueName) {
+		if (ProcessorType.EXTENDED.name().equals(processorType)){
+			sendExtendedMessages(messageList,largeMessageQueueName);
+		} else if (ProcessorType.S3.name().equals(processorType)) {
+			initS3Client(connectorClientAttributes);
+			messageList.getPutObjectRequest().forEach(message -> this.awsS3Client.putObject(message));
```

Comment:
Any reason we want to put the init methods in the sendMessages codepath?  I know it will see the client is already initialized but why even go into those methods every time we try and send a message instead of initializing on creation of the client?

Reply:
Yes, let's check this scenario: when appender is used with `largeMessageMode="S3"` it's not required to get `awsSQS` nor `awsExtended` instantiated to append logs into s3. So, as there's already this logic implemented at method `sendMessages`:

```

if (ProcessorType.EXTENDED.name().equals(processorType)){
 .. send to sqs-extended
} else if (ProcessorType.S3.name().equals(processorType)) {
 .. send to s3
} else {
 ... send to sqs
}
``` 
I took the chance to not replicate if-else logic to decide which client should be created. 

Hope it's clear enough, I really try to explain it as better as I could :D

Reply:
Resolved